sympy/solvers/solveset.py

```diff
@@ -978,6 +985,310 @@ def _solveset(f, symbol, domain, _check=False):
     return result
 
 
+def _solve_expo(f, symbol):
+    r"""
+    Helper function for solving (supported) exponential equations.
+
+    Exponential equations are the sum of (currently) at most
+    two terms with one or both of them having a power with a
+    symbol-dependent exponent.
+
+    For example
+
+    .. math:: 5^{2x + 3} - 5^{3x - 1}
+
+    .. math:: 4^{5 - 9x} - e^{2 - x}
+
+    The function reduces the equation (`f`) to a better log form
+    (if possible) and returns the modified equation which can be
+    further handled by `solveset`. `None` is returned if the equation
+    is not of supported type.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.solveset import _solve_expo as solve_expo
+    >>> from sympy import symbols
+    >>> x = symbols('x', real=True)
+    >>> solve_expo(3**(2*x) - 2**(x + 3), x)
+    2*x*log(3) - (x + 3)*log(2)
+
+    * Proof of correctness of the method
+
+    The logarithm function is the inverse of the exponential function.
+    The defining relation between exponentiation and logarithm is:
+
+    .. math:: {log_b x} = y \enspace if \enspace b^y = x
+
+    Therefore if we are given an equation with exponent terms, we can
+    convert every term to its corresponding log form. This is achieved by
+    taking logarithms and expanding the equation using log identities
+    so that it can easily be handled by `solveset`.
+
+    For example:
+
+    .. math:: 3^{2x} = 2^{x + 3}
+
+    Taking log both sides will reduce the equation to
+
+    .. math:: (2x)*log(3) = (x + 3)*log(2)
+
+    This form can be easily handed by `solveset`.
+    """
+
+    if not (isinstance(f, Add) and len(f.args) == 2):
+        # solving for the sum of more than two powers is possible
+        # but not yet implemented
+        return None
+
+    lhs, rhs = list(ordered(f.args))
+    log_type_equation = expand_log(log(lhs)) - expand_log(log(-rhs))
+
+    return log_type_equation
```

Comment:
What is going to happen when the user selected the complex domain for the solution in solveset? The symbol won't be real, will it? And then the equation won't simplify as desired. I think you will get an infinite loop. But if there is no solution with numbers having an imaginary part then the real solution should be returned...and somewhere that change to a real variable will have to take place. I suspect that when solving in the complex domain an attempt at the complex solution will have to be followed with an attempt at a real solution.

Reply:
> What is going to happen when the user selected the complex domain for the solution in solveset?

The equation won't get simplified (by `expand_log`) but still it will be transformed to `log` hence it won't fall into infinite loop.

As of now untill `_transolve` gets finalised, @aktech suggested to not focus on solving in complex domain. I guess once this PR gets into a proper shape, I can add solving in complex domain in a seperate PR. What do you say?

Reply:
Not solving in complex domain is fine, but it should not break with an infinite loop.